btk/draw_blends.py

```diff
@@ -187,8 +184,17 @@ def __next__(self):
         for s in self.surveys:
             pix_stamp_size = int(self.stamp_size / s.pixel_scale)
 
-            # create WCS and PSF information
-            psf = [get_psf(s, filt, self.atmospheric_model) for filt in s.filters]
+            # make PSF and WCS
+            psf = []
+            for filt in s.filters:
+                if callable(filt.psf):
+                    psf.append(filt.psf()) # generate the PSF with the provided function
```

Comment:
It might be good to check at some point that `filt.psf()` (the returned psf) is indeed a `galsim.GSObject` ?

Reply:
Yes indeed !